src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -1151,29 +1153,35 @@ int publish_reserve(const DoutPrefixProvider* dpp,
         // TODO: take default reservation size from conf
         constexpr auto DEFAULT_RESERVATION = 4 * 1024U;  // 4K
         res.size = DEFAULT_RESERVATION;
-        librados::ObjectWriteOperation op;
-        bufferlist obl;
-        int rval;
+        // librados::ObjectWriteOperation op;
+        // bufferlist obl;
+        // int rval;
         const auto& queue_name = topic_cfg.dest.persistent_queue;
         // TODO: Replace these with Redis
-        cls_2pc_queue_reserve(op, res.size, 1, &obl, &rval);
-        auto ret = rgw_rados_operate(
-            res.dpp, res.store->getRados()->get_notif_pool_ctx(), queue_name,
-            &op, res.yield, librados::OPERATION_RETURNVEC);
+        // cls_2pc_queue_reserve(op, res.size, 1, &obl, &rval);
+
+        auto ret = rgw::redisqueue::reserve(conn, queue_name, res.size, res.yield);
+        // auto ret = rgw_rados_operate(
+        //     res.dpp, res.store->getRados()->get_notif_pool_ctx(), queue_name,
+        //     &op, res.yield, librados::OPERATION_RETURNVEC);
         if (ret < 0) {
           ldpp_dout(res.dpp, 1)
               << "ERROR: failed to reserve notification on queue: "
               << queue_name << ". error: " << ret << dendl;
           // if no space is left in queue we ask client to slow down
           return (ret == -ENOSPC) ? -ERR_RATE_LIMITED : ret;
         }
-        ret = cls_2pc_queue_reserve_result(obl, res_id);
-        if (ret < 0) {
-          ldpp_dout(res.dpp, 1)
-              << "ERROR: failed to parse reservation id. error: " << ret
-              << dendl;
-          return ret;
-        }
+
+        // FIX ME: How is a reservation id used?
+        // Should we emulate this with Redis?
+
+        // ret = cls_2pc_queue_reserve_result(obl, res_id);
```

Comment:
how do you get the reservation id (`res_id`) without parsing the result of the `reserve()` call?

Reply:
There is no id associated with a reservation. 
We discussed there was no requirement for one if space reserved for all messages are same.